src/app/components/ElectronicDelivery/ElectronicDeliveryForm.jsx

```diff
@@ -24,22 +50,38 @@ class ElectronicDeliveryForm extends React.Component {
   submit(e) {
     e.preventDefault();
 
-    this.validate();
-    this.props.submitRequest(this.state);
+    if (validate(this.state.form, (error) => this.setState({ error }))) {
+      this.props.submitRequest(this.state);
+    }
   }
 
   handleUpdate(e, input) {
-    this.setState({ [input]: e.target.value });
-  }
-
-  validate() {
-    // Verify data
-    console.log(this.state);
+    // Kind of hard to read. Basically, the `form` property is being updated and all
+    // the values are being retained. If we don't `extend` the object value for `form`,
+    // then only the last value in the form gets updated and the rest are gone.
+    this.setState({ form: _extend(this.state.form, { [input]: e.target.value }) });
   }
 
   render() {
-    const errorClass = ''; //'nypl-field-error';
+    const errorClass = {
```

Comment:
I think 
`
{
         name: '',
         email: '',
        chapter: '',
         author: '',
          date: '',
         volume: '',
         issue: '',
          'starting-page': '',
         'ending-page': '',
        },
`

can be in default props as we use it a couple times in the component.